dali/kernels/test/test_tensors.h

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_TEST_TEST_TENSORS_H_
+#define DALI_KERNELS_TEST_TEST_TENSORS_H_
+
+#include <cuda_runtime_api.h>
+#include <dali/kernels/tensor_view.h>
+#include <memory>
+#include <utility>
+
+namespace dali {
+namespace kernels {
+
+template <typename T, int dim = DynamicDimensions>
+class TestTensorList {
+ public:
+  void reshape(TensorListShape<dim> shape) {
+    cpumem_.reset();
+    gpumem_.reset();
+    any_ = { nullptr, shape };
+  }
+
+  void invalidate_cpu() {
+    cpumem_.reset();
+  }
+
+  void invalidate_gpu() {
+    gpumem_.reset();
+  }
+
+  template <int out_dim = dim>
+  TensorListView<StorageCPU, T, out_dim> cpu(cudaStream_t stream = 0) {
+    TensorListView<StorageCPU, T, out_dim> ret;
+    if (!cpumem_) {
+      auto size = any_.num_elements() * sizeof(T);
+      char *ptr = new char[size];
```

Comment:
why not allocate with std::vector<> ?

Reply:
Because it's owned by a `unique_ptr`. That's mostly for uniformity with GPU version.